app/src/input_manager.c

```diff
@@ -911,31 +906,31 @@ sc_input_manager_process_mouse_wheel(struct sc_input_manager *im,
 
 static void
 sc_input_manager_process_gamepad_device(struct sc_input_manager *im,
-                                       const SDL_ControllerDeviceEvent *event) {
-    if (event->type == SDL_CONTROLLERDEVICEADDED) {
-        SDL_GameController *gc = SDL_GameControllerOpen(event->which);
+                                       const SDL_GamepadDeviceEvent *event) {
+    if (event->type == SDL_EVENT_GAMEPAD_ADDED) {
+        SDL_Gamepad *gc = SDL_OpenGamepad(event->which);
         if (!gc) {
             LOGW("Could not open game controller");
             return;
         }
 
-        SDL_Joystick *joystick = SDL_GameControllerGetJoystick(gc);
+        SDL_Joystick *joystick = SDL_GetGamepadJoystick(gc);
         if (!joystick) {
             LOGW("Could not get controller joystick");
-            SDL_GameControllerClose(gc);
+            SDL_CloseGamepad(gc);
             return;
         }
```

Comment:
Same things as in previous comments: reword "game controller" and "controller" to "gamepad" and find a better name for the `gc` pointer.